slackscot.go

```diff
@@ -299,6 +299,13 @@ func OptionTestMode(terminationCh chan bool) Option {
 	}
 }
 
+//OptionCommandPrefix sets a prefix to all commands that is used instead of at-mentioning the bot
+func OptionCommandPrefix(cmdPrefix string) Option {
```

Comment:
I think that's the right user-facing interface but I don't love hijacking the `selfUserPrefix` to hold the cmd prefix override. I know this could add a lot of overhead to what's currently only a few lines but here's an approach that could be more elegant:

* Add a new interface
```
type CommandMatcher interface {
   IsCmd(msg slack.Msg) bool
   UsagePrefix() string
}
```

* Move the self identify fields (`selfID, selfBotID, selfName, selfUserPrefix`) to a struct `selfIdentity` and have that struct implement the new interface:
```
func (self *selfIdentity) IsCmd(msg slack.Msg) bool {
    return strings.HasPrefix(msg.Text, self.userPrefix)
}

func (self *selfIdentify) UsagePrefix() {
     return ""
}
```

* Have a new `cmdMatcher` field in `Slackscot` which is going to get initialized to the `new(selfIdentity)` in the constructor and potentially overridden by `OptionCommandPrefix`.

* Implement `OptionCommandPrefix` to something like:
```
func OptionCommandPrefix(cmdPrefix string) Option {
	return func(s *Slackscot) {		
		s.cmdMatcher = prefixedCommand{prefix: cmdPrefix}
	}
}

type prefixedCommand struct {
	prefix string
}

func (pc *prefixedCommand) IsCmd(msg slack.Msg) bool {
    return strings.HasPrefix(msg.Text, pc.prefix)
}

func (pc *prefixedCommand) UsagePrefix() string {
    return pc.prefix
}
```

* Remove the current function `IsCommand` and replace by `s.cmdMatcher.IsCmd(m)`. To do that, you'll have migrate the `isDirectMessage` functionality to another function (i.e. `if isCommand, isDirectMsg := s.cmdMatcher.IsCmd(m), isDirectMessage(m); isCommand {`

* Change the `help` plugin to take in the `CommandMatcher` so it can do `UsagePrefix()` on it to generate the correct help. 

I know, it's more work, right? I can help if that's overwhelming. You'll also have to add tests to confirm that the commands are correctly triggered using the new option and that the help is generated with the correct message.